src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -1,70 +1,184 @@
   isXs,
   relay,
   viewer,
 }) => {
+  const tracking = useTracking()
   const { t } = useTranslation()
-  const [value, setValue] = useState("")
+  const [value, setValue] = useState(getSearchTerm(window.location))
+  const [selectedPill, setSelectedPill] = useState<PillType>(TOP_PILL)
+  const [searchIncrement, setSearchIncrement] = useState(0)
 
   const options = extractNodes(viewer.searchConnection)
-  const formattedOptions = options.map(option => {
-    return {
-      text: option.displayLabel!,
-      value: option.displayLabel!,
-      subtitle: getLabel({
-        displayType: option.displayType ?? "",
+  const formattedOptions: SuggionItemOptionProps[] = options.map(
+    (option, index) => {
+      return {
+        text: option.displayLabel!,
+        value: option.displayLabel!,
+        subtitle:
+          getLabel({
+            displayType: option.displayType ?? "",
+            typename: option.__typename,
+          }) ?? "",
+        imageUrl: option.imageUrl!,
+        showArtworksButton: !!option.statuses?.artworks,
+        showAuctionResultsButton: !!option.statuses?.auctionLots,
+        href: option.href!,
         typename: option.__typename,
-      }),
-      imageUrl: option.imageUrl!,
-      showArtworksButton: !!option.statuses?.artworks,
-      showAuctionResultsButton: !!option.statuses?.auctionLots,
-      href: option.href!,
+        item_number: index,
+        item_type: option.displayType!,
+      }
     }
-  })
+  )
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value)
+  useUpdateEffect(() => {
+    tracking.trackEvent({
+      action_type:
+        options.length > 0
+          ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+          : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+      query: value,
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchIncrement])
+
+  const refetch = (value: string, entity?: SearchEntity) => {
+    const entities = entity ? [entity] : []
+    const performanceStart = performance && performance.now()
 
-    //TODO: Add throttle
     relay.refetch(
       {
         hasTerm: true,
-        term: event.target.value,
+        term: value,
+        entities: entities,
       },
       null,
       error => {
         if (error) {
           logger.error(error)
           return
         }
+
+        if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+          reportPerformanceMeasurement(performanceStart)
+        }
+
+        setSearchIncrement(prevCounter => prevCounter + 1)
       }
     )
   }
+  const throttledSearchRequest = useMemo(
+    () => throttle(refetch, SEARCH_THROTTLE_DELAY, { leading: true }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value)
+    throttledSearchRequest(event.target.value)
```

Comment:
we should not refetch data if the user has entered **less** than 2 characters (as [here](https://github.com/artsy/force/blob/f85fdad385bcfae7c7fdeb016a2f49e59fa42bad/src/Components/Search/NewSearch/NewSearchBarInput.tsx#L187))

Reply:
we will also have to cancel throttled search request if the user **has cleared** the search value/query or entered less than 2 characters

https://user-images.githubusercontent.com/3513494/236203551-53e77ae2-a629-420a-9bfc-6821b66fb596.mp4

Reply:
Seems for me that it's better to use debouncing instead of throttling in order to avoid firing search requests for previously typed strings. I also decreased debounce interval from 500 to 250, this way user experience feels more responsive, but we also do not fire extra requests if user types very fast.

Not sure if we need to dispose previous refetch requests. I think user has to be like incredibly fast clearing the input field, no?

Reply:
> it's better to use debouncing instead of throttling

maybe you're right. but it seems to me that when using trolling, the UX will be more responsive and friendly (since the user will see the "fresh"/"relevant" search results for the query more often. for example, like Amazon or Wikipedia)